src/main/java/com/powsybl/openloadflow/util/BranchState.java

```diff
@@ -0,0 +1,66 @@
+package com.powsybl.openloadflow.util;
+
+import com.powsybl.openloadflow.network.*;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class BranchState {
+    private final double a1;
+    private final double r1;
+    private final double x;
+    private final double r;
+    private final DiscretePhaseControl phaseControl;
+    private final DiscreteVoltageControl voltageControl;
+
+    public BranchState(LfBranch b) {
+        PiModel piModel = b.getPiModel();
```

Comment:
Can you start with just `a1` and `r1` ? For the controls, it will be more complex (and we don't know how to do).